src/gui/EditWidgetIcons.cpp

```diff
@@ -314,43 +317,76 @@ void EditWidgetIcons::addCustomIconFromFile()
         QString filter = QString("%1 (%2);;%3 (*)").arg(tr("Images"),
                     Tools::imageReaderFilter(), tr("All files"));
 
-        QString filename = QFileDialog::getOpenFileName(
-                    this, tr("Select Image"), "", filter);
-        if (!filename.isEmpty()) {
-            auto icon = QImage(filename);
-            if (!icon.isNull()) {
-                addCustomIcon(QImage(filename));
+        auto filenames = QFileDialog::getOpenFileNames(this, tr("Select Image(s)"), "", filter);
+        if (!filenames.empty()) {
+            QStringList errornames;
+            int numexisting = 0;
+            for (const auto& filename : filenames) {
+                if (!filename.isEmpty()) {
+                    auto icon = QImage(filename);
+                    if (!icon.isNull()) {
+                        if (!addCustomIcon(QImage(filename))) {
+                            ++numexisting;
+                        }
+                    } else {
+                        errornames << filename;
+                    }
+                }
+            }
+
+            int numloaded = filenames.size() - errornames.size() - numexisting;
+            QString msg;
+
+            if (numloaded > 0) {
+                msg = tr("Successfully loaded %1 of %2 icons").arg(numloaded).arg(filenames.size());
+            } else {
+                msg = tr("No icons were loaded");
+            }
+
+            if (numexisting > 0) {
+                msg += ", " + tr("%1 icons already existed").arg(numexisting);
```

Comment:
This doesn't work. You cannot assume other languages to append clauses in this order or use commas to separate them. This needs to be two separate messages.